--- sexp.h
+++ sexp.h
@@ -390,6 +390,9 @@
   sexp_uint_t size, max_size, chunk_size;
   sexp_free_list free_list;
   sexp_heap next;
+  puchi_host host;
+  puchi_module_ops modules;
+  int modules_set;
   /* note this must be aligned on a proper heap boundary, */
   /* so we can't just use char data[] */
   char *data;
@@ -1732,7 +1735,7 @@
 SEXP_API void sexp_add_static_libraries(struct sexp_library_entry_t* libraries);
 
 SEXP_API sexp sexp_alloc_tagged_aux(sexp ctx, size_t size, sexp_uint_t tag sexp_current_source_param);
-SEXP_API sexp sexp_make_context(sexp ctx, size_t size, size_t max_size);
+SEXP_API sexp sexp_make_context(sexp ctx, size_t size, size_t max_size, const puchi_host *host);
 SEXP_API sexp sexp_cons_op(sexp ctx, sexp self, sexp_sint_t n, sexp head, sexp tail);
 SEXP_API sexp sexp_list2(sexp ctx, sexp a, sexp b);
 SEXP_API sexp sexp_list3(sexp ctx, sexp a, sexp b, sexp c);
@@ -1892,7 +1895,7 @@
 #if ! SEXP_USE_BOEHM && ! SEXP_USE_MALLOC
 SEXP_API void sexp_gc_init (void);
 SEXP_API int sexp_grow_heap (sexp ctx, size_t size, size_t chunk_size);
-SEXP_API sexp_heap sexp_make_heap (size_t size, size_t max_size, size_t chunk_size);
+SEXP_API sexp_heap sexp_make_heap (const puchi_host *host, size_t size, size_t max_size, size_t chunk_size);
 SEXP_API void sexp_mark (sexp ctx, sexp x);
 SEXP_API sexp sexp_sweep (sexp ctx, size_t *sum_freed_ptr);
 #if SEXP_USE_FINALIZERS
